player_state.h

```diff
@@ -1,6 +1,8 @@
 #ifndef PLAYER_STATE_H
 #define PLAYER_STATE_H
 
+#include <iostream>
```

Comment:
I predict this include is the one that is the best of your knowledge. Pro-tip: use `#include <iosfwd>`. I will accept this regardless :-)

Reply:
if include `iosfwd` instead of `iostream`, then `cout` will not be in `std::`, so that `std::cout << ps;` will fail. How to solve this issue?

Reply:
I removed std::cout << ps, I assume this test is not important and for saving compiling time it would be better to stick to `iosfwd`

Reply:
@EvoLandEco my bad: in header files, use `iosfwd`, in implementation (`.cpp`) files, include `iostream`. This has to do with forward declarations (whatever those are :-) ).

The test you changed **is** important; it proes that a `player_state` can be streamed. No worries, I/you will create an Issue to put it back in :-)

Reply:
I have put it back, with suggested `iosfwd` and `iostream` usage

Reply:
@EvoLandEco awesome :+1: